astarte_hwid.c

```diff
@@ -37,13 +37,18 @@ astarte_err_t astarte_hwid_get_id(uint8_t *hardware_id)
     int embedded_flash = (chip_info.features & CHIP_FEATURE_EMB_FLASH) != 0;
     int bluetooth = (chip_info.features & CHIP_FEATURE_BT) != 0;
     int ble = (chip_info.features & CHIP_FEATURE_BLE) != 0;
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
+    uint16_t revision = chip_info.revision / 100u;
+#else
+    uint16_t revision = chip_info.revision;
+#endif
```

Comment:
Maybe add a comment explaining what's happening and possibly pointing to the changelog of esp-idf, because it's hard to understand why this `ifdef` is needed without any kind of context while reading the code

Reply:
I think anyone reading the macro would understand that this change relates to an update to IDF version 5.0.
But sure, adding a comment won't hurt.